### PR TITLE
Update Superflow

### LINKED ISSUE #14: Update Superflow
Over the recent months, we've changed the way we handle releases.

We no longer use the `master` branch, instead now building AppStore builds directly from release branch tags, and testing them in TestFlight.

Superflow should be updated, both the diagram and the text, appropriately.

### PR DESCRIPTION
**Pull request by amulware**
Closes #14

This mostly updates the release flow, but also addresses a few minor things and attempts to clean up the long document a little.

I hope you like how much simpler the diagram has become. I certainly do.

### CODE REVIEW

File: superflow.md [hunk 1]
```diff
@@ -85,48 +76,32 @@ Our apps follows the following versioning scheme:
 - The `minor` component is incremented by one as the first commit of every release branch **(and in no other case)**. The same commit also removes the `maintenance` component.
 - The `maintenance` is added and incremented by one as the first commit of every hot fix branch **(and in no other case)**.
 
-In addition, there is also the build number which is not visible to the public and only used to distinguish different builds internally. This number is incremented in all the three cases above. In addition it is incremented for any additional pre-release build created.
+In addition, there is also the build number which is not visible to the public and only used to distinguish different builds internally. This number is incremented automatically for most of our apps. If it is not, it should be incremented for each created release tag.
 
 In essence, the versioning scheme can thus be thought of as:
 
-    major.release[.hotfix]
+    major.release[.hot_fix]
```
Comment (Odrakir) [inline superflow.md]: major.release[.hot_fix] (build number) ?

File: superflow.md [hunk 2]
```diff
@@ -12,68 +12,59 @@ Below you will find Vincent's original diagram adapted and extended correspondin
...
 ### Release workflow
 
-The above explanations of SuperFlow are not only sub-set of allowed operations, but are in fact exhaustive. This means that there are no other valid ways to create releases, than outlined above.
+The above explanations of SuperFlow are not only a sub-set of allowed operations, but are in fact exhaustive. This means that there are no other valid ways to create releases, than outlined above.
 
 In summary:
 - Release and hot fix branches are two only two ways of creating new public releases.
-- Release and hot fix branches branch from `develop` and `master` respectively.
-- Release and hot fix branches are always merged into both `master` and `develop` on completion. This merge is always performed using a non-fast-forward, non-squashed merge commit.
+- Release and hot fix branches branch from `develop` and the latest release tag respectively.
```
Comment (Odrakir) [inline superflow.md]: In the diagram, the hotfix branches out from a release branch and not develop.

File: superflow.md [hunk 3]
```diff
@@ -85,48 +76,32 @@ Our apps follows the following versioning scheme:
...
-- If there are any bugs, PRs will be made against this branch, creating a new prerelease once they are merged, testing resumes then.
-- If everything works, create a PR for that branch into master, add the complete, correct and accumulated change log of all changes since the last release (not fixes to bugs introduced in this release)
-- When approved, merge but don't squash, also don´t delete the branch.
-- Create a relase in GH on master with the tag "*.*.*". Bitrise will build automatically. The changelog should be the same as in the PR.
-- Create another PR from the same branch into develop. When approved merge but don't squash.
-- Upload the bitrise build to the App Store / Play Store.
+From the above it follows that these are the steps to release a new version of the app:
+
+- Create a release branch from develop. This contains all the changes wanted in this new version.
+- Commit the new version to the release branch.
```
Comment (Odrakir) [inline superflow.md]: Is this line clear? It took me a couple of reads.

File: superflow.md (hunk 2 above)
Comment (amulware) [reply superflow.md]: Hotfix branches branch from the latest release tag, as it says here and is shown in the diagram. Which part is not clear?

Comment (kornelijepetak) [reply superflow.md]: That's what the sentence says:
Release branches from develop,
Fix branches from latest release tag.
Hence the word `respectively` :)

File: superflow.md (hunk 1 above)
Comment (amulware) [reply superflow.md]: Why bother mentioning it, especially since we automated it for iOS and soon will for android as well?

File: superflow.md (hunk 3 above)
Comment (amulware) [reply superflow.md]: Yeah, this could be clearer.

File: superflow.md (hunk 1 above)
Comment (Odrakir) [reply superflow.md]: It's part of the version number.

Comment (amulware) [reply superflow.md]: It's mentioned in the paragraph above, which I think is sufficient. This scheme is only trying to explain the semantic meaning of the public part of the version. The build number doesn't have much of a semantic meaning for us and is just a technical thing we have to observe.
